lib/src/main/kotlin/bstrees/balancers/RBBalancer.kt

```diff
@@ -3,8 +3,86 @@ package bstrees.balancers
 import bstrees.nodes.RBNode
 
 class RBBalancer<T : Comparable<T>> : TreeBalancer<T, RBNode<T>> {
-    override fun balanceAfterInsertion(node: RBNode<T>): RBNode<T> {
-        TODO("Not yet implemented")
+    private fun color(node: RBNode<T>?) = node?.color ?: RBNode.Color.Black
+
+    private fun isRed(node: RBNode<T>?) = color(node) == RBNode.Color.Red
+
+    private fun isBlack(node: RBNode<T>?) = !isRed(node)
```

Comment:
<img width="557" alt="image" src="https://user-images.githubusercontent.com/15161335/229357067-410b26fc-c48c-4b36-a96d-65f1060cd2fc.png">

Reply:
This function may be needed further

Reply:
ok